src/api/app/controllers/webui/request_controller.rb

```diff
@@ -310,9 +305,11 @@ def require_request
     redirect_back(fallback_location: user_show_path(User.current)) && return
   end
 
-  def get_target_package_maintainers(actions)
-    actions = actions.uniq { |action| action[:tpkg] }
-    actions.flat_map { |action| Package.find_by_project_and_name(action[:tprj], action[:tpkg]).try(:maintainers) }.compact.uniq
+  def target_package_maintainers
+    distinct_bs_request_actions = @bs_request.bs_request_actions.select(:target_project, :target_package).distinct
```

Comment:
we know and we are working on that. any suggestion?

Reply:
well, to be honest I am not sure what of them is right... I was just pointing it out and expecting that you give me the reasoning why this can be changed.

> any suggestion?

I guess you mean to keep the old behavior but in an SQL query (more efficient than before). You can do this with an SQL subquery.

Reply:
concrete suggestion? theory we know all :)

Reply:
We noticed, but I think that's alright. Actually it should even be better this way since it's does no longer consider packages with the same name but different project as the same.
But it's true that we could split this out to another commit, or PR. I don't really mind.

Reply:
@vpereira 

> concrete suggestion? theory we know all :)

SQL subquery is a concrete suggestion.  I guess you know how to do this in Rails. Please if not, let me know and I can send you some good documentation. :smile: 

@bgeuken 

> We noticed, but I think that's alright. Actually it should even be better this way since it's does no longer consider packages with the same name but different project as the same.
But it's true that we could split this out to another commit, or PR. I don't really mind.

I tend to think you are right, but if this was wrong before it should have introduced bugs....

Reply:
> I tend to think you are right, but if this was wrong before it should have introduced bugs....

Well, it's only used for visual information (listing package maintainers and showing a hint in the decision box when there are any). And having two request actions with different target projects but the same package name is quite an edge case.

But now we cover it :bowtie: